File: frontend/src/app/business/dtos/book.request.ts
import {AuthorResponse} from "./author.response";
import {OnlyId} from "./OnlyId";

export class BookRequest {
  private _name: string;
  private _isbn: string;
  private _quantity: number;
  private _authors: OnlyId[];
  private _publisher: OnlyId;
  private _literatureCategory: OnlyId;

  public constructor() {
  }

  get name(): string {
    return this._name;
  }

  set name(value: string) {
    this._name = value;
  }

  get isbn(): string {
    return this._isbn;
  }

  set isbn(value: string) {
    this._isbn = value;
  }

  get quantity(): number {
    return this._quantity;
  }

  set quantity(value: number) {
    this._quantity = value;
  }

  get authorsId(): number[] {
    return this._authors.map(u => u.id);
  }

  set authorsIds(authorsId: number[]) {
    this._authors = authorsId.map(id => new OnlyId(id));
  }

  get publisherId(): number {
    return this._publisher.id;
  }

  set publisherId(value: number) {
    this._publisher = new OnlyId(value);
  }

  get literatureCategoryId(): number {
    return this._literatureCategory.id;
  }

  set literatureCategoryId(value: number) {
    this._literatureCategory = new OnlyId(value);
  }
}

File: backend/src/main/java/com/mendes/library/model/DTO/BookDTO/BookRequest.java
package com.mendes.library.model.DTO.BookDTO;

import com.mendes.library.model.DTO.AuthorDTO.AuthorResponse;
import com.mendes.library.model.DTO.LiteratureCategoryDTO.LiteratureCategoryDTO;
import com.mendes.library.model.DTO.PublisherDTO.PublisherDTO;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import javax.validation.constraints.NotNull;
import java.util.List;

@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
public class BookRequest {
    private String name;
    @NotNull(message = "The isbn is required.")
    private String isbn;
    @NotNull(message = "The author is required.")
    private List<AuthorResponse> authors;
    @NotNull(message = "The publisher is required.")
    private PublisherDTO publisher;
    @NotNull(message = "The quantity is required.")
    private Integer quantity;
    @NotNull(message = "The category name is required.")
    private LiteratureCategoryDTO categoryName;

}
